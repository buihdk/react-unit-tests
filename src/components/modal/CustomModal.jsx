import React from 'react';
import PropTypes from 'prop-types';

import { Modal } from 'antd';
import { noop } from 'src/utils';

const CustomModal = props => {
  const {
    loading,
    handleCancelModal,
    modalVisible,
    children,
    title,
    handleSubmit,
    okText,
    cancelText,
    closable,
    customWidth,
  } = props;

  return (
    <Modal
      title={title}
      width={customWidth}
      visible={modalVisible}
      onOk={handleSubmit}
      onCancel={handleCancelModal}
      okText={okText}
      confirmLoading={loading}
      closable={closable}
      cancelText={cancelText}
    >
      {children}
    </Modal>
  );
};

CustomModal.propTypes = {
  loading: PropTypes.bool,
  modalVisible: PropTypes.bool,
  children: PropTypes.shape({}),
  title: PropTypes.string,
  okText: PropTypes.string,
  cancelText: PropTypes.string,
  closable: PropTypes.bool,
  customWidth: PropTypes.number,
  handleCancelModal: PropTypes.func,
  handleSubmit: PropTypes.func,
};

CustomModal.defaultProps = {
  loading: false,
  modalVisible: false,
  children: {},
  title: '',
  okText: '',
  cancelText: '',
  closable: false,
  customWidth: 400,
  handleCancelModal: noop,
  handleSubmit: noop,
};

export default CustomModal;
