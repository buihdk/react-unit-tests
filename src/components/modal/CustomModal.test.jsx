import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CustomModal from './CustomModal';

describe('CustomModal', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<CustomModal />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
