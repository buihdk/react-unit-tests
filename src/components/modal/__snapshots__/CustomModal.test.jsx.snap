// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal renders without crashing 1`] = `
<Modal
  cancelText=""
  closable={false}
  confirmLoading={false}
  maskTransitionName="fade"
  okText=""
  okType="primary"
  onCancel={[Function]}
  onOk={[Function]}
  title=""
  transitionName="zoom"
  visible={false}
  width={400}
>
  <Component />
</Modal>
`;
